pandas/tests/indexes/common.py

```diff
@@ -342,20 +343,48 @@ def test_copy_and_deepcopy(self, indices):
         new_copy = indices.copy(deep=True, name="banana")
         assert new_copy.name == "banana"
 
-    def test_duplicates(self, indices):
+    def test_has_duplicates(self, indices):
         if type(indices) is not self._holder:
-            return
+            pytest.skip('Can only check if we have the correct type')
         if not len(indices) or isinstance(indices, MultiIndex):
-            return
+            # MultiIndex tested separately in:
+            # tests/indexes/multi/test_unique_and_duplicates
+            pytest.skip('Skip check for empty Index and MultiIndex')
+
         idx = self._holder([indices[0]] * 5)
         assert not idx.is_unique
         assert idx.has_duplicates
 
+    @pytest.mark.parametrize('keep', ['first', 'last', False])
+    def test_duplicated(self, indices, keep):
+        if type(indices) is not self._holder:
```

Comment:
can you use isinstance here

Reply:
@jreback `isinstance` of what? I copied the checks from other tests, because I didn't know all the different cases that flow into `common.py`.

Reply:
isinstande of self._holder, is this what we do elsewhere?

Reply:
not in `indexes/common.py`, as far as I can see. There's several `isinstance` of course, but never for `self._holder`, which can apparently be `None` as well.

Reply:
@jreback If I change
```
if type(indices) is not self._holder:
```
to
```
if not isinstance(indices, self._holder):
```
I get 9 failures (instead of skips), all of which are from

`tests/indexes/test_numeric.TestInt64Index`, when run against `DatetimeIndex`, `PeriodIndex` or `TimedeltaIndex`, mainly because of `TypeError: Unsafe NumPy casting, you must explicitly cast` it seems.

Reply:
@jreback Seem to have found something: in my normal workbook, I get
```
pd.__version__
# '0.24.0.dev0+321.g0fe6ded52.dirty'
isinstance(pd.PeriodIndex, pd.Int64Index)
# False
```
but for some reason, within `tests/indexes/common.py`, this is the opposite:
```
    @pytest.mark.parametrize('keep', ['first', 'last', False])
    def test_duplicated(self, indices, keep):
        if not isinstance(indices, self._holder):
            pytest.skip('Can only check if we know the index type')
        if not len(indices) or isinstance(indices, (MultiIndex)):
            # MultiIndex tested separately in:
            # tests/indexes/multi/test_unique_and_duplicates
            pytest.skip('Skip check for empty Index and MultiIndex')
        if isinstance(indices, (PeriodIndex, DatetimeIndex)):
>           raise ValueError(f'{type(indices).__name__}, {self._holder.__name__}, {isinstance(indices, self._holder)}')
E           ValueError: PeriodIndex, Int64Index, True
```
Same happens for `DatetimeIndex`.

Reply:
Opened a follow-up: #22211

Reply:
Nevermind, I had gotten confused between instances and classes.

So `DatetimeIndex`, `PeriodIndex` and `TimedeltaIndex` are subclasses of `Int64Index` - but are unsafe to cast to Int64...? I guess this is intentional?

In any case, under these circumstances, I'm even more convinced that it's best to just stay with the `if type(indices) is not self._holder:` condition.